osu.Game/Screens/Edit/Editor.cs

```diff
@@ -1077,6 +1094,69 @@ private void seekControlPoint(int direction)
                 clock.Seek(found.Time);
         }
 
+        private void seekHitObject(int direction)
+        {
+            var found = direction < 1
+                ? editorBeatmap.HitObjects.LastOrDefault(p => p.StartTime < clock.CurrentTimeAccurate)
+                : editorBeatmap.HitObjects.FirstOrDefault(p => p.StartTime > clock.CurrentTimeAccurate);
+
+            if (found != null)
+                clock.SeekSmoothlyTo(found.StartTime);
+        }
+
+        [CanBeNull]
+        public event Action<double> ShowSampleEditPopoverRequested;
```

Comment:
This flow is a bit ugly. Probably fine maybe but still ugly.

Move `public` things to top of file and document though.